core/server/master/src/main/java/alluxio/master/metastore/rocks/RocksBlockMetaStore.java

```diff
@@ -79,6 +85,20 @@ public class RocksBlockMetaStore implements BlockMetaStore {
   private final AtomicReference<ColumnFamilyHandle> mBlockMetaColumn = new AtomicReference<>();
   private final AtomicReference<ColumnFamilyHandle> mBlockLocationsColumn = new AtomicReference<>();
   private final LongAdder mSize = new LongAdder();
+  /**
+   * When this is true, stop serving all requests as the RocksDB is being closed.
+   * This flag serves two purposes:
+   * 1. Before an r/w operation, if this is observed then bail the operation early.
+   * 2. During an r/w operation like iteration, if this is observed then abort the iteration.
+   */
+  private final AtomicBoolean mClosed = new AtomicBoolean(false);
+  /**
+   * The RWLock is used to guarantee thread safety between r/w and RocksDB closing/restarting.
+   * When the RocksDB is closing/restarting, no concurrent r/w should be present.
+   * Therefore, we acquire the write lock on closing/restarting, and acquire a read lock for
+   * reading/writing the RocksDB.
+   */
+  private final ReadWriteLock mDbStateLock = new ReentrantReadWriteLock();
```

Comment:
These two come in pairs. So i think i can just wrap them into an object like `RocksProtocol`? Then
```
abstract class RocksProtocol {
  final AtomicBoolean mClosed;
  final ReadWriteLock mDbStateLock;
  abstract LockResource checkAndAcquireReadLock();
  abstract void abortCheck();
}

class RocksBlockMetaStore implements BlockMetaStore extends RocksProtocol {
...
}
```